narwhals/_arrow/utils.py

```diff
@@ -204,24 +272,18 @@ def broadcast_and_extract_native(
             fill_value = lhs[0]
             if backend_version < (13,) and hasattr(fill_value, "as_py"):
                 fill_value = fill_value.as_py()
-            left_result = pa.chunked_array(
-                [
-                    pa.array(
-                        np.full(shape=rhs.len(), fill_value=fill_value),
-                        type=lhs._native_series.type,
-                    )
-                ]
+            arr = pa.array(
+                np.full(shape=rhs.len(), fill_value=fill_value),
+                type=lhs._native_series.type,
             )
-            return left_result, rhs._native_series
+            return chunked_array(arr), rhs._native_series
         return lhs._native_series, rhs._native_series
     return lhs._native_series, rhs
 
 
 def broadcast_and_extract_dataframe_comparand(
-    length: int,
-    other: Any,
-    backend_version: tuple[int, ...],
-) -> Any:
+    length: int, other: ArrowSeries, backend_version: tuple[int, ...]
+) -> pa.BooleanArray:
```

Comment:
🤔 this doesn't have to be boolean, right? plenty of binary operations don't return booleans?